frontend/src/components/TagForm/TagForm.js

```diff
@@ -0,0 +1,39 @@
+import { useMutate } from "../../hooks/use-mutate"
+import { useRef } from "react"
+import { useTags } from "../../hooks/tag-hooks"
+import { useParams } from "react-router"
+
+const TagForm = () => {
+    const formRef = useRef()
+    const { updateTag } = useMutate()
+    const { addTag } = useMutate()
+    const { id } = useParams()
+    const { data } = useTags()
```

Comment:
ditto on the `useTag` since this is pulling from a single source. A get-all is an expensive operation that shouldn't be run unless necessary. 

When using `useTag`, GQL expects inputs as objects. The object should be similar to how you would use graphiql to run a test on your query.